### DISCUSSION TITLE
Database ConnectorError

### DISCUSSION THREAD

**Discussion by David-orovs** [vercel/community#4629] [Help]
### Summary

I am having an issue querying my db from my application hosted in vercel, i get
 PrismaClientUnknownRequestError: 
```
Invalid `prisma.user.findFirst()` invocation:


Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Error { kind: Db, cause: Some(DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState(E26000), message: "prepared statement \"PGBOUNCER_1\" does not exist", detail: None, hint: None, position: None, where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("prepare.c"), line: Some(451), routine: Some("FetchPreparedStatement") }) }), transient: false })
```
 when i try to login, meaning it is not seeing a db to findFirst from user, it works well on my local db, please help solve this issue

### Example

an api call to my api shows internal server error, cos i dont return the exact error

### Steps to Reproduce

All my connections on prisma are correct, it worked well until an hour ago

**Comment by evaldast** [comment]
I believe I am facing the same issue.

https://github.com/orgs/vercel/discussions/4630

Using non-Pooler `POSTGRES_PRISMA_URL` seems to fix the issue.

**Reply by David-orovs** [reply]
@evaldast what do you mean by non-Pooler POSTGRES_PRISMA_URL

**Reply by evaldast** [reply]
working host example:
`ep-raspy-sun-746185.eu-central-1.postgres.vercel-storage.com`

pgBouncer (non working host)
`ep-raspy-sun-746185-pooler.eu-central-1.postgres.vercel-storage.com`

If I switch back from Pooler (current Vercel default) to old default, it works.

**Reply by David-orovs** [reply]
@evaldast mine is working back, I did not even switch, it was not working before now it is working

**Reply by David-orovs** [reply]
i think it is linked to this https://github.com/orgs/vercel/discussions/4631 what do you think @evaldast

**Comment by IgorGala** [comment]
Also having the same issue, but it wasn't the case just a few days before. Is non-pooler the only option here?

**Reply by David-orovs** [reply]
@Wodzu2137 mine is the same except postgres.vercel-storage.com:5432 part od the url, the one on vercel env does not have the port

**Comment by David-orovs** [comment]
@amyegan please help

**Comment by evaldast** [comment]
Setting `directUrl` in Prisma's `datasource` seems to work as a workaround (has to be non PgBouncer/Pooler host). `POSTGRES_URL_NON_POOLING` can be used.

From Prisma docs:
https://www.prisma.io/docs/guides/performance-and-optimization/connection-management/configure-pg-bouncer#prisma-migrate-and-pgbouncer-workaround

**Comment by amyegan** [comment]
Thanks for reporting this. It was related to a Postgres service degradation incident which has been resolved. You can find more info on the incident report at https://www.vercel-status.com/incidents/mz0ghb4sdxsm

Please let me know if you're still having trouble! 🙏

**Reply by David-orovs** [reply]
@amyegan yes I am still having an issue here https://github.com/orgs/vercel/discussions/4631
